haystack/nodes/prompt/prompt_node.py

```diff
@@ -347,6 +348,9 @@ def invoke(self, *args, **kwargs):
             if stop_words:
                 sw = StopWordsCriteria(tokenizer=self.pipe.tokenizer, stop_words=stop_words)
                 model_input_kwargs["stopping_criteria"] = StoppingCriteriaList([sw])
+            if top_k:
+                model_input_kwargs["num_return_sequences"] = top_k
+                model_input_kwargs["num_beams"] = top_k
```

Comment:
Handling in `HFInvocationLayer`

Reply:
In AnswerGenerator we warn if `num_beams` < `num_return_sequences`. If the user wants to control them separately, she can do so via `model_kwargs`, but I can't even imagine why one would do so.